// @aws-sdk/client-bedrock-agent/dist-es/models/enums.js
export const CustomControlMethod = {
    RETURN_CONTROL: "RETURN_CONTROL",
};
export const ActionGroupState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const Type = {
    ARRAY: "array",
    BOOLEAN: "boolean",
    INTEGER: "integer",
    NUMBER: "number",
    STRING: "string",
};
export const RequireConfirmation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const ActionGroupSignature = {
    AMAZON_CODEINTERPRETER: "AMAZON.CodeInterpreter",
    AMAZON_USERINPUT: "AMAZON.UserInput",
    ANTHROPIC_BASH: "ANTHROPIC.Bash",
    ANTHROPIC_COMPUTER: "ANTHROPIC.Computer",
    ANTHROPIC_TEXTEDITOR: "ANTHROPIC.TextEditor",
};
export const AgentCollaboration = {
    DISABLED: "DISABLED",
    SUPERVISOR: "SUPERVISOR",
    SUPERVISOR_ROUTER: "SUPERVISOR_ROUTER",
};
export const AgentStatus = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    NOT_PREPARED: "NOT_PREPARED",
    PREPARED: "PREPARED",
    PREPARING: "PREPARING",
    UPDATING: "UPDATING",
    VERSIONING: "VERSIONING",
};
export const MemoryType = {
    SESSION_SUMMARY: "SESSION_SUMMARY",
};
export const OrchestrationType = {
    CUSTOM_ORCHESTRATION: "CUSTOM_ORCHESTRATION",
    DEFAULT: "DEFAULT",
};
export const CreationMode = {
    DEFAULT: "DEFAULT",
    OVERRIDDEN: "OVERRIDDEN",
};
export const PromptState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const PromptType = {
    KNOWLEDGE_BASE_RESPONSE_GENERATION: "KNOWLEDGE_BASE_RESPONSE_GENERATION",
    MEMORY_SUMMARIZATION: "MEMORY_SUMMARIZATION",
    ORCHESTRATION: "ORCHESTRATION",
    POST_PROCESSING: "POST_PROCESSING",
    PRE_PROCESSING: "PRE_PROCESSING",
};
export const AgentAliasStatus = {
    CREATING: "CREATING",
    DELETING: "DELETING",
    DISSOCIATED: "DISSOCIATED",
    FAILED: "FAILED",
    PREPARED: "PREPARED",
    UPDATING: "UPDATING",
};
export const AliasInvocationState = {
    ACCEPT_INVOCATIONS: "ACCEPT_INVOCATIONS",
    REJECT_INVOCATIONS: "REJECT_INVOCATIONS",
};
export const RelayConversationHistory = {
    DISABLED: "DISABLED",
    TO_COLLABORATOR: "TO_COLLABORATOR",
};
export const KnowledgeBaseState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const DataDeletionPolicy = {
    DELETE: "DELETE",
    RETAIN: "RETAIN",
};
export const CrawlFilterConfigurationType = {
    PATTERN: "PATTERN",
};
export const ConfluenceAuthType = {
    BASIC: "BASIC",
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
};
export const ConfluenceHostType = {
    SAAS: "SAAS",
};
export const SalesforceAuthType = {
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
};
export const SharePointAuthType = {
    OAUTH2_CLIENT_CREDENTIALS: "OAUTH2_CLIENT_CREDENTIALS",
    OAUTH2_SHAREPOINT_APP_ONLY_CLIENT_CREDENTIALS: "OAUTH2_SHAREPOINT_APP_ONLY_CLIENT_CREDENTIALS",
};
export const SharePointHostType = {
    ONLINE: "ONLINE",
};
export const DataSourceType = {
    CONFLUENCE: "CONFLUENCE",
    CUSTOM: "CUSTOM",
    REDSHIFT_METADATA: "REDSHIFT_METADATA",
    S3: "S3",
    SALESFORCE: "SALESFORCE",
    SHAREPOINT: "SHAREPOINT",
    WEB: "WEB",
};
export const WebScopeType = {
    HOST_ONLY: "HOST_ONLY",
    SUBDOMAINS: "SUBDOMAINS",
};
export const ChunkingStrategy = {
    FIXED_SIZE: "FIXED_SIZE",
    HIERARCHICAL: "HIERARCHICAL",
    NONE: "NONE",
    SEMANTIC: "SEMANTIC",
};
export const EnrichmentStrategyMethod = {
    CHUNK_ENTITY_EXTRACTION: "CHUNK_ENTITY_EXTRACTION",
};
export const ContextEnrichmentType = {
    BEDROCK_FOUNDATION_MODEL: "BEDROCK_FOUNDATION_MODEL",
};
export const StepType = {
    POST_CHUNKING: "POST_CHUNKING",
};
export const ParsingModality = {
    MULTIMODAL: "MULTIMODAL",
};
export const ParsingStrategy = {
    BEDROCK_DATA_AUTOMATION: "BEDROCK_DATA_AUTOMATION",
    BEDROCK_FOUNDATION_MODEL: "BEDROCK_FOUNDATION_MODEL",
};
export const DataSourceStatus = {
    AVAILABLE: "AVAILABLE",
    DELETE_UNSUCCESSFUL: "DELETE_UNSUCCESSFUL",
    DELETING: "DELETING",
};
export const FlowConnectionType = {
    CONDITIONAL: "Conditional",
    DATA: "Data",
};
export const SupportedLanguages = {
    PYTHON_3: "Python_3",
};
export const PerformanceConfigLatency = {
    OPTIMIZED: "optimized",
    STANDARD: "standard",
};
export const RerankingMetadataSelectionMode = {
    ALL: "ALL",
    SELECTIVE: "SELECTIVE",
};
export const VectorSearchRerankingConfigurationType = {
    BEDROCK_RERANKING_MODEL: "BEDROCK_RERANKING_MODEL",
};
export const CachePointType = {
    DEFAULT: "default",
};
export const ConversationRole = {
    ASSISTANT: "assistant",
    USER: "user",
};
export const PromptTemplateType = {
    CHAT: "CHAT",
    TEXT: "TEXT",
};
export const FlowNodeInputCategory = {
    EXIT_LOOP: "ExitLoop",
    LOOP_CONDITION: "LoopCondition",
    RETURN_VALUE_TO_LOOP_START: "ReturnValueToLoopStart",
};
export const FlowNodeIODataType = {
    ARRAY: "Array",
    BOOLEAN: "Boolean",
    NUMBER: "Number",
    OBJECT: "Object",
    STRING: "String",
};
export const FlowNodeType = {
    AGENT: "Agent",
    COLLECTOR: "Collector",
    CONDITION: "Condition",
    INLINE_CODE: "InlineCode",
    INPUT: "Input",
    ITERATOR: "Iterator",
    KNOWLEDGE_BASE: "KnowledgeBase",
    LAMBDA_FUNCTION: "LambdaFunction",
    LEX: "Lex",
    LOOP: "Loop",
    LOOP_CONTROLLER: "LoopController",
    LOOP_INPUT: "LoopInput",
    OUTPUT: "Output",
    PROMPT: "Prompt",
    RETRIEVAL: "Retrieval",
    STORAGE: "Storage",
};
export const FlowStatus = {
    FAILED: "Failed",
    NOT_PREPARED: "NotPrepared",
    PREPARED: "Prepared",
    PREPARING: "Preparing",
};
export const ConcurrencyType = {
    AUTOMATIC: "Automatic",
    MANUAL: "Manual",
};
export const IncompatibleLoopNodeType = {
    COLLECTOR: "Collector",
    CONDITION: "Condition",
    INPUT: "Input",
    ITERATOR: "Iterator",
};
export const FlowValidationSeverity = {
    ERROR: "Error",
    WARNING: "Warning",
};
export const FlowValidationType = {
    CYCLIC_CONNECTION: "CyclicConnection",
    DUPLICATE_CONDITION_EXPRESSION: "DuplicateConditionExpression",
    DUPLICATE_CONNECTIONS: "DuplicateConnections",
    INCOMPATIBLE_CONNECTION_DATA_TYPE: "IncompatibleConnectionDataType",
    INVALID_LOOP_BOUNDARY: "InvalidLoopBoundary",
    LOOP_INCOMPATIBLE_NODE_TYPE: "LoopIncompatibleNodeType",
    MALFORMED_CONDITION_EXPRESSION: "MalformedConditionExpression",
    MALFORMED_NODE_INPUT_EXPRESSION: "MalformedNodeInputExpression",
    MISMATCHED_NODE_INPUT_TYPE: "MismatchedNodeInputType",
    MISMATCHED_NODE_OUTPUT_TYPE: "MismatchedNodeOutputType",
    MISSING_CONNECTION_CONFIGURATION: "MissingConnectionConfiguration",
    MISSING_DEFAULT_CONDITION: "MissingDefaultCondition",
    MISSING_ENDING_NODES: "MissingEndingNodes",
    MISSING_LOOP_CONTROLLER_NODE: "MissingLoopControllerNode",
    MISSING_LOOP_INPUT_NODE: "MissingLoopInputNode",
    MISSING_NODE_CONFIGURATION: "MissingNodeConfiguration",
    MISSING_NODE_INPUT: "MissingNodeInput",
    MISSING_NODE_OUTPUT: "MissingNodeOutput",
    MISSING_STARTING_NODES: "MissingStartingNodes",
    MULTIPLE_LOOP_CONTROLLER_NODES: "MultipleLoopControllerNodes",
    MULTIPLE_LOOP_INPUT_NODES: "MultipleLoopInputNodes",
    MULTIPLE_NODE_INPUT_CONNECTIONS: "MultipleNodeInputConnections",
    UNFULFILLED_NODE_INPUT: "UnfulfilledNodeInput",
    UNKNOWN_CONNECTION_CONDITION: "UnknownConnectionCondition",
    UNKNOWN_CONNECTION_SOURCE: "UnknownConnectionSource",
    UNKNOWN_CONNECTION_SOURCE_OUTPUT: "UnknownConnectionSourceOutput",
    UNKNOWN_CONNECTION_TARGET: "UnknownConnectionTarget",
    UNKNOWN_CONNECTION_TARGET_INPUT: "UnknownConnectionTargetInput",
    UNKNOWN_NODE_INPUT: "UnknownNodeInput",
    UNKNOWN_NODE_OUTPUT: "UnknownNodeOutput",
    UNREACHABLE_NODE: "UnreachableNode",
    UNSATISFIED_CONNECTION_CONDITIONS: "UnsatisfiedConnectionConditions",
    UNSPECIFIED: "Unspecified",
};
export const IngestionJobStatus = {
    COMPLETE: "COMPLETE",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    STARTING: "STARTING",
    STOPPED: "STOPPED",
    STOPPING: "STOPPING",
};
export const IngestionJobFilterAttribute = {
    STATUS: "STATUS",
};
export const IngestionJobFilterOperator = {
    EQ: "EQ",
};
export const IngestionJobSortByAttribute = {
    STARTED_AT: "STARTED_AT",
    STATUS: "STATUS",
};
export const SortOrder = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
export const ContentDataSourceType = {
    CUSTOM: "CUSTOM",
    S3: "S3",
};
export const DocumentStatus = {
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    DELETING: "DELETING",
    FAILED: "FAILED",
    IGNORED: "IGNORED",
    INDEXED: "INDEXED",
    IN_PROGRESS: "IN_PROGRESS",
    METADATA_PARTIALLY_INDEXED: "METADATA_PARTIALLY_INDEXED",
    METADATA_UPDATE_FAILED: "METADATA_UPDATE_FAILED",
    NOT_FOUND: "NOT_FOUND",
    PARTIALLY_INDEXED: "PARTIALLY_INDEXED",
    PENDING: "PENDING",
    STARTING: "STARTING",
};
export const InlineContentType = {
    BYTE: "BYTE",
    TEXT: "TEXT",
};
export const CustomSourceType = {
    IN_LINE: "IN_LINE",
    S3_LOCATION: "S3_LOCATION",
};
export const MetadataValueType = {
    BOOLEAN: "BOOLEAN",
    NUMBER: "NUMBER",
    STRING: "STRING",
    STRING_LIST: "STRING_LIST",
};
export const MetadataSourceType = {
    IN_LINE_ATTRIBUTE: "IN_LINE_ATTRIBUTE",
    S3_LOCATION: "S3_LOCATION",
};
export const RedshiftProvisionedAuthType = {
    IAM: "IAM",
    USERNAME: "USERNAME",
    USERNAME_PASSWORD: "USERNAME_PASSWORD",
};
export const RedshiftServerlessAuthType = {
    IAM: "IAM",
    USERNAME_PASSWORD: "USERNAME_PASSWORD",
};
export const RedshiftQueryEngineType = {
    PROVISIONED: "PROVISIONED",
    SERVERLESS: "SERVERLESS",
};
export const IncludeExclude = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
export const RedshiftQueryEngineStorageType = {
    AWS_DATA_CATALOG: "AWS_DATA_CATALOG",
    REDSHIFT: "REDSHIFT",
};
export const QueryEngineType = {
    REDSHIFT: "REDSHIFT",
};
export const KnowledgeBaseType = {
    KENDRA: "KENDRA",
    SQL: "SQL",
    VECTOR: "VECTOR",
};
export const EmbeddingDataType = {
    BINARY: "BINARY",
    FLOAT32: "FLOAT32",
};
export const SupplementalDataStorageLocationType = {
    S3: "S3",
};
export const KnowledgeBaseStorageType = {
    MONGO_DB_ATLAS: "MONGO_DB_ATLAS",
    NEPTUNE_ANALYTICS: "NEPTUNE_ANALYTICS",
    OPENSEARCH_MANAGED_CLUSTER: "OPENSEARCH_MANAGED_CLUSTER",
    OPENSEARCH_SERVERLESS: "OPENSEARCH_SERVERLESS",
    PINECONE: "PINECONE",
    RDS: "RDS",
    REDIS_ENTERPRISE_CLOUD: "REDIS_ENTERPRISE_CLOUD",
    S3_VECTORS: "S3_VECTORS",
};
export const KnowledgeBaseStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETE_UNSUCCESSFUL: "DELETE_UNSUCCESSFUL",
    DELETING: "DELETING",
    FAILED: "FAILED",
    UPDATING: "UPDATING",
};
